### PR TITLE
refactor: add browser version of package and write docs for it

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- Add webpack configuration for browser version of package
- Add tests for browser version
- Update Readme.md
- Update source code for better treeshaking - especially use directly supported JSON Schemas

cc @smoya 

**Related issue(s)**
Part of #481 
Part of #482

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -17,4 +14,4 @@ export const xParserCircularProps = 'x-parser-circular-props';
 export const EXTENSION_REGEX = /^x-[\w\d.\-_]+$/;
 
 // Only >=2.0.0 versions are supported
-export const specVersions = Object.keys(specs).filter((version: string) => !['1.0.0', '1.1.0', '1.2.0', '2.0.0-rc1', '2.0.0-rc2'].includes(version));
\ No newline at end of file
+export const specVersions = ['2.0.0', '2.1.0', '2.2.0', '2.3.0', '2.4.0'];
```
Comment (smoya) [inline src/constants.ts:17]: Why this rollback to the hardcoded version? 🤯

File: src/schema-parser/asyncapi-schema-parser.ts [hunk 2]
```diff
@@ -9,6 +7,13 @@ import type { AsyncAPISchema, SchemaValidateResult } from '../types';
 import type { SchemaParser, ParseSchemaInput, ValidateSchemaInput } from '../schema-parser';
 import type { v2 } from '../spec-types';
 
+// import only 2.X.X AsyncAPI JSON Schemas for better treeshaking
+import * as asyncAPI2_0_0Schema from '@asyncapi/specs/schemas/2.0.0.json';
+import * as asyncAPI2_1_0Schema from '@asyncapi/specs/schemas/2.1.0.json';
+import * as asyncAPI2_2_0Schema from '@asyncapi/specs/schemas/2.2.0.json';
+import * as asyncAPI2_3_0Schema from '@asyncapi/specs/schemas/2.3.0.json';
+import * as asyncAPI2_4_0Schema from '@asyncapi/specs/schemas/2.4.0.json';
```
Comment (smoya) [inline src/schema-parser/asyncapi-schema-parser.ts:15]: This is not scalable at all. I don't want to keep adding lines here every time we release a version.

File: src/schema-parser/asyncapi-schema-parser.ts [hunk 3]
```diff
@@ -98,3 +103,20 @@ function preparePayloadSchema(asyncapiSchema: v2.AsyncAPISchemaDefinition, versi
     definitions
   };
 }
+
+function getSchema(version: string): Record<string, unknown> {
```
Comment (smoya) [inline src/schema-parser/asyncapi-schema-parser.ts:107]: Similar to https://github.com/asyncapi/parser-js/pull/596/files#r966432839

File: src/constants.ts [hunk 4]
```diff
@@ -16,5 +15,5 @@ export const xParserCircularProps = 'x-parser-circular-props';
 
 export const EXTENSION_REGEX = /^x-[\w\d.\-_]+$/;
 
-// Only >=2.0.0 versions are supported
-export const specVersions = Object.keys(specs).filter((version: string) => !['1.0.0', '1.1.0', '1.2.0', '2.0.0-rc1', '2.0.0-rc2'].includes(version));
\ No newline at end of file
+export const specVersions = Object.keys(specs);
```
Comment (smoya) [inline src/constants.ts:18]: Do we now allow all versions?

Comment (magicmatatjahu) [reply src/constants.ts:18]: No, look at this PR https://github.com/asyncapi/spec-json-schemas/pull/248 We need to reduce the size of `specs` package. Now, I will switch to the old code and then I will create new PR in this repo, when we will merge mentioned PR.

Comment (smoya) [reply src/constants.ts]: Ok, I see you changed the code after. 👍

**Review by smoya — APPROVED** [review]
LGTM!

**Pull request merged**